src/Symfony/Component/Runtime/SymfonyRuntime.php

```diff
@@ -0,0 +1,253 @@
+use Symfony\Component\ErrorHandler\Debug;
+use Symfony\Component\ErrorHandler\ErrorHandler;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\HttpKernel\TerminableInterface;
+use Symfony\Component\Runtime\Internal\MissingDotenv;
+use Symfony\Component\Runtime\Internal\MissingErrorHandler;
+
+// Help opcache.preload discover always-needed symbols
+class_exists(MissingDotenv::class, false) || class_exists(Dotenv::class) || class_exists(MissingDotenv::class);
+class_exists(MissingErrorHandler::class, false) || class_exists(ErrorHandler::class) || class_exists(MissingErrorHandler::class);
+
+/**
+ * Knows the basic conventions to run Symfony apps.
+ *
+ * Accepts two options:
+ *  - "debug" to toggle debugging features;
+ *  - "env" to define the name of the environment the app runs in.
+ *
+ * When these options are not defined, they will fallback:
+ *  - to reading the "APP_DEBUG" and "APP_ENV" environment variables;
+ *  - to parsing the "--env|-e" and "--no-debug" command line arguments
+ *    if the "symfony/console" component is installed.
+ *
+ * When the "symfony/dotenv" component is installed, .env files are loaded.
+ * When "symfony/error-handler" is installed, it is used to improve error handling.
+ *
+ * On top of the base arguments provided by the parent runtime,
+ * this runtime can feed the app-closure with arguments of type:
+ *  - Request from "symfony/http-foundation" if the component is installed;
+ *  - Application, Command, InputInterface and/or OutputInterface
+ *    from "symfony/console" if the component is installed.
+ *
+ * This runtime can handle app-closures that return instances of either:
+ *  - HttpKernelInterface,
+ *  - Response,
+ *  - Application,
+ *  - Command,
+ *  - or Closure(): int
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class SymfonyRuntime extends BaseRuntime
+{
+    private $request;
+    private $input;
+    private $output;
+    private $application;
+    private $command;
+    private $env;
+
+    public function __construct(array $options = [])
+    {
+        $this->env = $options['env'] ?? null;
+        $_SERVER['APP_ENV'] = $options['env'] ?? $_SERVER['APP_ENV'] ?? null;
+
+        parent::__construct(['debug' => $options['debug'] ?? true]);
+
+        if (isset($_SERVER['argv']) && null === $this->env && class_exists(ArgvInput::class)) {
+            $this->getInput();
+        }
+
+        if (class_exists(MissingDotenv::class, false)) {
+            $_SERVER['APP_DEBUG'] = filter_var($options['debug'] ?? $_SERVER['APP_DEBUG'] ?? true, FILTER_VALIDATE_BOOLEAN) ? '1' : '0';
+        } else {
+            (new Dotenv())->bootEnv(\dirname($_SERVER['APP_VENDOR_DIR'] ?? \dirname(__DIR__, 2)).'/.env');
+        }
+
+        if (class_exists(MissingErrorHandler::class, false)) {
+            // no-op
+        } elseif ($_SERVER['APP_DEBUG']) {
+            umask(0000);
```

Comment:
Is this umask still necessary? We can now use `symfony serve` for development

And on production you may **not** want the files in var/cache to become readable for everybody

Reply:
I think we should keep it to maximise compatiblity yes.

Reply:
With the current symfony recipes it's easy to edit / remove this line to adapt it to your project needs. 

Here it is not anymore...

Reply:
I never cared about this line; can you please explain when and why it's important to change this ?

Reply:
We don't need it during development. So we always simply removed this line to avoid accidentally creating files on a shared production server that are readable for everybody with `bin/console -e dev` for example.

Setting up APP_DEBUG=0 on a production server also avoids this and we do this already. 

So I'm ok with keeping this line.